data/serializer/src/main/java/tech/pegasys/teku/api/response/v2/beacon/GetFinalizedCheckpointStateResponse.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright ConsenSys Software Inc., 2022
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.api.response.v2.beacon;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import tech.pegasys.teku.api.schema.Version;
+import tech.pegasys.teku.api.schema.altair.BeaconStateAltair;
+import tech.pegasys.teku.api.schema.bellatrix.BeaconStateBellatrix;
+import tech.pegasys.teku.api.schema.interfaces.State;
+import tech.pegasys.teku.api.schema.phase0.BeaconStatePhase0;
+
+@SuppressWarnings("JavaCase")
+public class GetFinalizedCheckpointStateResponse {
```

Comment:
as far as i can tell this is identical to `GetStateResponseV2`, i'd be tempted to just use that unless there's a reason to publish the same object definition again?

Reply:
I just removed the `execution_optimistic` flag from the `GetFinalizedCheckpointStateResponse`. We're discussing if we should keep it (It that case I'll use the existing `GetStateResponseV2`) or simply refuse to return a state if it's optimistically syncing.